lib/Backend/IRBuilder.cpp

```diff
@@ -2303,12 +2303,35 @@ void
 IRBuilder::BuildReg4(Js::OpCode newOpcode, uint32 offset, Js::RegSlot dstRegSlot, Js::RegSlot src1RegSlot,
                     Js::RegSlot src2RegSlot, Js::RegSlot src3RegSlot)
 {
-    IR::Instr *     instr;
-    Assert(newOpcode == Js::OpCode::Concat3);
+    IR::Instr *     instr = nullptr;
+    Assert(newOpcode == Js::OpCode::Concat3 || newOpcode == Js::OpCode::Restify);
 
     IR::RegOpnd * src1Opnd = this->BuildSrcOpnd(src1RegSlot);
     IR::RegOpnd * src2Opnd = this->BuildSrcOpnd(src2RegSlot);
-    IR::RegOpnd * src3Opnd = this->BuildSrcOpnd(src3RegSlot);
+    IR::RegOpnd * src3Opnd = this->BuildSrcOpnd(src3RegSlot);    
+
+    if (newOpcode == Js::OpCode::Restify)
+    {
+        IR::RegOpnd * src0Opnd = this->BuildSrcOpnd(dstRegSlot);
+        instr = IR::Instr::New(Js::OpCode::ExtendArg_A, IR::RegOpnd::New(TyVar, m_func), src3Opnd, m_func);
+        this->AddInstr(instr, offset);
+
+        instr = IR::Instr::New(Js::OpCode::ExtendArg_A, IR::RegOpnd::New(TyVar, m_func), src2Opnd, instr->GetDst(), m_func);
+        this->AddInstr(instr, Js::Constants::NoByteCodeOffset);
+
+        instr = IR::Instr::New(Js::OpCode::ExtendArg_A, IR::RegOpnd::New(TyVar, m_func), src1Opnd, instr->GetDst(), m_func);
+        this->AddInstr(instr, Js::Constants::NoByteCodeOffset);
+
+        instr = IR::Instr::New(Js::OpCode::ExtendArg_A, IR::RegOpnd::New(TyVar, m_func), src0Opnd, instr->GetDst(), m_func);
+        this->AddInstr(instr, Js::Constants::NoByteCodeOffset);
+
+        IR::Opnd *prevInstr = instr->GetDst();
```

Comment:
>prevInstr [](start = 18, length = 9)

nit: could use a better name

Reply:
what about `firstArgLoc`?

---
In reply to: [204578281](https://github.com/Microsoft/ChakraCore/pull/5498#discussion_r204578281) [](ancestors = 204578281)

Reply:
just `firstArg` would do for me. Thanks

---
In reply to: [204619741](https://github.com/Microsoft/ChakraCore/pull/5498#discussion_r204619741) [](ancestors = 204619741,204578281)